signer/src/transaction_signer.rs

```diff
@@ -1012,85 +1030,159 @@ where
+            &mut FrostCoordinator,
+            &UnsignedMockTransaction,
+        ),
+        Error,
+    > {
+        let state_machine_id = StateMachineId::RotateKey(aggregate_key, *bitcoin_chain_tip);
 
-                self.wsts_frost_state_machines
-                    .get_or_insert_mut(state_machine_id, || coordinator);
-            }
+        if !self.wsts_frost_state_machines.contains(&state_machine_id) {
+            let storage = self.context.get_storage();
+            let coordinator =
+                Self::create_frost_coordinator(&storage, aggregate_key, self.signer_private_key)
+                    .await?;
+            self.wsts_frost_state_machines
+                .put(state_machine_id, coordinator);
         }
 
-        Ok(state_machine_id)
+        let state_machine = self
+            .wsts_frost_state_machines
+            .get_mut(&state_machine_id)
+            .ok_or(Error::MissingFrostStateMachine(Box::new(aggregate_key)))?;
+
+        let mock_tx = UnsignedMockTransaction::new(aggregate_key.into());
+        let mock_tx = self
+            .wsts_frost_mock_txs
+            .get_or_insert(state_machine_id, || mock_tx);
+
+        Ok((state_machine_id, state_machine, mock_tx))
     }
 
     #[tracing::instrument(skip_all)]
     async fn handle_rotate_key_message(
         &mut self,
-        aggregate_key: PublicKeyXOnly,
+        bitcoin_chain_tip: &model::BitcoinBlockHash,
         id: StateMachineId,
         msg: &WstsNetMessage,
     ) -> Result<(), Error> {
+        // We should only be handling messages for the rotate-key state machine.
+        // We'll grab the aggregate key from the id as well.
+        let aggregate_key = match id {
+            StateMachineId::RotateKey(aggregate_key, _) => aggregate_key,
+            _ => {
+                tracing::warn!(
+                    "🔐 unexpected state machine id for pre-rotate-key validation signing round"
+                );
+                return Err(Error::UnexpectedStateMachineId(Box::new(id)));
+            }
+        };
+
         let state_machine = self.wsts_frost_state_machines.get_mut(&id);
         let Some(state_machine) = state_machine else {
-            tracing::warn!("missing frost coordinator for pre-rotate-key validation");
+            tracing::warn!("🔐 missing frost coordinator for pre-rotate-key validation");
             return Err(Error::MissingFrostStateMachine(Box::new(aggregate_key)));
         };
 
-        tracing::info!(?msg, "processing frost coordinator message");
+        tracing::trace!(?msg, "🔐 processing frost coordinator message");
 
         let (_, result) = state_machine.process_message(msg)?;
 
         match result {
-            Some(OperationResult::SignSchnorr(_)) => {
-                tracing::info!("successfully completed pre-rotate-key validation signing round");
-                self.wsts_frost_results.put(id, true);
+            Some(OperationResult::SignSchnorr(sig) | OperationResult::SignTaproot(sig)) => {
```

Comment:
Is `SignSchnorr` still relevant?

Reply:
Think I just left it since it's what's done in the coordinator, but we're not requesting a `SignatureType::Schnorr` anymore so it can probably be removed without issue.

Reply:
I had removed it somewhere along the way to #1301, so it's gone